From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Fenix - Control Gecko Password Manager

If the password manager is enabled/disabled from the browser's UI settings, this ensures that the functionality is also fully enabled/disabled
within Gecko.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/SavedLoginsSettingFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/SavedLoginsSettingFragment.kt
index a553816882ca..c6dd82723108 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/SavedLoginsSettingFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/SavedLoginsSettingFragment.kt
@@ -16,6 +16,8 @@ import org.mozilla.fenix.settings.RadioButtonPreference
 import org.mozilla.fenix.settings.SharedPreferenceUpdater
 import org.mozilla.fenix.settings.requirePreference
 import org.mozilla.fenix.utils.view.addToRadioGroup
+import org.ironfoxoss.ironfox.utils.GeckoSettingsBridge
+import org.mozilla.fenix.ext.requireComponents
 
 /** Settings screen allowing users to choose whether newly used passwords should be remembered or not. */
 class SavedLoginsSettingFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFragment {
@@ -41,6 +43,7 @@ class SavedLoginsSettingFragment : PreferenceFragmentCompat(), SystemInsetsPadde
                             Logins.SaveLoginsSettingChangedExtra(Setting.ASK_TO_SAVE.name)
                         )
                     }
+                    GeckoSettingsBridge.setPasswordManagerEnabled(requireContext(), requireComponents.core.engine)
                     // We want to reload the current session here so we can try to fill the current page
                     context?.components?.useCases?.sessionUseCases?.reload?.invoke()
                     return super.onPreferenceChange(preference, newValue)
@@ -59,6 +62,7 @@ class SavedLoginsSettingFragment : PreferenceFragmentCompat(), SystemInsetsPadde
                             Logins.SaveLoginsSettingChangedExtra(Setting.NEVER_SAVE.name)
                         )
                     }
+                    GeckoSettingsBridge.setPasswordManagerEnabled(requireContext(), requireComponents.core.engine)
                     // We want to reload the current session here so we don't save any currently inserted login
                     context?.components?.useCases?.sessionUseCases?.reload?.invoke()
                     return super.onPreferenceChange(preference, newValue)
-- 
